stencils/pace/stencils/update_atmos_state.py

```diff
@@ -220,57 +254,66 @@ def __init__(
+
+        self._apply_physics_to_dycore = ApplyPhysicsToDycore(
             stencil_factory,
             grid_data,
             self.namelist,
             comm,
             grid_info,
         )
+        self._dycore_only = dycore_only
+        self._apply_tendencies = apply_tendencies
 
     def __call__(
         self,
         dycore_state,
         phy_state,
+        tendency_state,
         dt: float,
     ):
-        self._fill_GFS(
-            phy_state.prsi, phy_state.physics_updated_specific_humidity, 1.0e-9
-        )
-        self._prepare_tendencies_and_update_tracers(
-            self._u_dt,
-            self._v_dt,
-            self._pt_dt,
-            phy_state.physics_updated_ua,
-            phy_state.physics_updated_va,
-            phy_state.physics_updated_pt,
-            phy_state.physics_updated_specific_humidity,
-            phy_state.physics_updated_qliquid,
-            phy_state.physics_updated_qrain,
-            phy_state.physics_updated_qsnow,
-            phy_state.physics_updated_qice,
-            phy_state.physics_updated_qgraupel,
-            phy_state.ua,
-            phy_state.va,
-            phy_state.pt,
-            dycore_state.qvapor,
-            dycore_state.qliquid,
-            dycore_state.qrain,
-            dycore_state.qsnow,
-            dycore_state.qice,
-            dycore_state.qgraupel,
-            phy_state.prsi,
-            dycore_state.delp,
-            self._rdt,
-        )
-        self._apply_physics2dycore(
-            dycore_state, self._u_dt, self._v_dt, self._pt_dt, dt=dt
-        )
+        if self._dycore_only:
+            self._fill_GFS_delp(dycore_state.delp, dycore_state.qvapor, 1.0e-9)
+        else:
+            self._fill_GFS_delp(
+                dycore_state.delp, phy_state.physics_updated_specific_humidity, 1.0e-9
+            )
+            self._prepare_tendencies_and_update_tracers(
+                tendency_state.u_dt,
+                tendency_state.v_dt,
+                tendency_state.pt_dt,
+                phy_state.physics_updated_ua,
+                phy_state.physics_updated_va,
+                phy_state.physics_updated_pt,
+                phy_state.physics_updated_specific_humidity,
+                phy_state.physics_updated_qliquid,
+                phy_state.physics_updated_qrain,
+                phy_state.physics_updated_qsnow,
+                phy_state.physics_updated_qice,
+                phy_state.physics_updated_qgraupel,
+                phy_state.ua,
+                phy_state.va,
+                phy_state.pt,
+                dycore_state.qvapor,
+                dycore_state.qliquid,
+                dycore_state.qrain,
+                dycore_state.qsnow,
+                dycore_state.qice,
+                dycore_state.qgraupel,
+                phy_state.prsi,
+                dycore_state.delp,
+                self._rdt,
+            )
+        if self._apply_tendencies:
```

Comment:
If apply_tendencies=False, it means we aren't running physics or fv_subgridz. If that's the case, does any code in this module need to run? The answer is probably "yeah, the fortran does a couple things in that case" - if that's the answer, please add a `# comment` talking about what it's doing, since it's fairly unexpected.

Reply:
I can add a comment